serializer_impl.hh

```diff
@@ -499,9 +499,11 @@ public:
 
     [[gnu::always_inline]]
     operator managed_bytes() && {
-        managed_bytes v(managed_bytes::initialized_later(), _stream.size());
-        _stream.read(reinterpret_cast<char*>(v.begin()), _stream.size());
-        return v;
+        managed_bytes mb(managed_bytes::initialized_later(), _stream.size());
+        for (managed_bytes_mutable_view v = mb; !v.empty(); v.remove_current()) {
```

Comment:
Okay, I will add the iterator.

Reply:
You can make it independent of any concrete view type, and just make it usable with any `FragmentedView`:
```c++
template <FragmentedView View>
struct fragment_range_iterator;

template <FragmentedView View>
struct fragment_range {
    View v;
    fragment_range_iterator<View> begin();
    fragment_range_iterator<View> end();
};

// So it can be used like this:
for (auto& fragment : fragment_range(my_view)) {
}
```

Reply:
Done. Take a look: https://github.com/scylladb/scylla/pull/7820/commits/b4eb358713168761d62e11aaeea330824c631768